lib/nfs_common.pm

```diff
@@ -183,5 +183,85 @@ sub yast2_server_initial {
     } while (not match_has_tag('nfs-config'));
 }
 
+sub install_service {
+    # Make sure packages are installed
+    zypper_call 'in yast2-nfs-server nfs-kernel-server', timeout => 480, exitcode => [0, 106, 107];
+}
+
+sub config_service {
+    my ($rw, $ro) = @_;
+
+    try_nfsv2();
+
+    prepare_exports($rw, $ro);
+
+    my $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'nfs-server');
+
+    yast2_server_initial();
+
+    # Start server
+    send_key 'alt-s';
+
+    # Disable NFSv4
+    send_key 'alt-v';
+    wait_still_screen 1;
+
+    yast_handle_firewall();
+
+    # Next step
+    send_key 'alt-n';
+
+    assert_screen 'nfs-overview';
+
+    add_shares($rw, $ro);
+
+    send_key 'alt-f';
+    wait_serial("$module_name-0") or die "'yast2 $module_name' didn't finish";
+
+}
+
+sub start_service {
+    my ($rw, $ro) = @_;
+    # Back on the console, test mount locally
+    clear_console;
+
+    # Server is up and running, client can use it now!
+    script_run "( journalctl -fu nfs-server > /dev/$serialdev & )";
+    check_nfs_ready($rw, $ro);
+}
+
+sub check_service {
+    my ($rw, $ro) = @_;
+
+    # Check NFS version
+    if ((is_sle('=12-sp1') || is_sle('=12-sp2')) && script_run 'nfsstat -m') {
+        record_soft_failure 'bsc#1140731';
+    }
+    else {
+        validate_script_output "nfsstat -m", sub { m/vers=3/ };
+    }
+
+    assert_script_run "exportfs | grep '${rw}\\|${ro}'";
+    assert_script_run "cat /etc/exports | tr -d ' \\t\\r' | grep '${rw}\\*(rw,\\|${ro}\\*(ro,'";
+    assert_script_run "cat /proc/fs/nfsd/exports";
+
+    assert_script_run('systemctl is-enabled nfs-server');
+    assert_script_run('systemctl is-active nfs-server');
+
+}
+
+sub check_y2_nfs_func {
+    my ($stage) = @_;
+    my $rw      = '/srv/nfs';
+    my $ro      = '/srv/ro';
+    $stage //= '';
+    if ($stage eq 'before') {
+        install_service();
```

Comment:
Those three lines and the hardcoded paths are now duplicated across two files, can that be improved?